src/main/java/seedu/address/model/RecipeBook.java

```diff
@@ -69,16 +69,14 @@ public void addRecipe(Recipe toAdd) {
     }
 
     /**
-     * Removes a recipe from the recipe book with its {@code recipeId}
+     * Removes a recipe from the recipe book
      */
-    public void removeRecipe(int recipeId) {
-        for (Recipe recipe : this.recipeList) {
-            if (recipe.getId() == recipeId) {
-                this.recipeList.remove(recipe);
-                return;
-            }
+    public void removeRecipe(Recipe recipe) {
+        try {
+            this.recipeList.remove(recipe);
+        } catch (RecipeNotFoundException e) {
+            throw new RecipeNotFoundException();
```

Comment:
It should be fine not catching the exception here considering the exception was thrown in the previous level. It's a runtime exception so we don't necessarily need to catch it, same as other runtime exceptions

Reply:
I think throwing an exception is fine so that the code can catch and display the corresponding error if needed